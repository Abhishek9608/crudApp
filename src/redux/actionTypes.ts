export enum ActionTypes {
  ADD_POST = "ADD_POST",
  DELETE_POST = "DELETE_POST",
  UPDATE_POST = "UPDATE_POST",
  GET_POSTS = "GET_POSTS"
}
