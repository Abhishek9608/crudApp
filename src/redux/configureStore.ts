import thunkMiddleware from "redux-thunk";

import {
  createStore, 
  applyMiddleware, 
} from "redux";

import reducers from "./reducer";

export const configureStore = () => {
  const store = createStore(
    reducers,
    applyMiddleware(thunkMiddleware)
  );

  return store;
};
