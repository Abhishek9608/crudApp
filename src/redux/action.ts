import { AppThunk } from "./index";
import {ActionTypes} from './actionTypes'
import { data } from "../types";

export const fetchPosts = () : AppThunk  =>{
  return (dispatch, getState) => {
    fetch("https://jsonplaceholder.typicode.com/posts")
      .then((response) => response.json())
      .then((json) => {
        console.log(json)
        dispatch({
          type: ActionTypes.GET_POSTS,
          posts: json
        })
      });
  };
}

export const addPost = (data : data): AppThunk => {
  return (dispatch) => {
    fetch('https://jsonplaceholder.typicode.com/posts', {
    method: 'POST',
    body: JSON.stringify(data)
  })
    .then((response) => response.json())
    .then((json) =>{
      dispatch(fetchPosts())
    });
  }
}

export const editPost = (data : data, id:number | undefined): AppThunk => {
  return (dispatch) => {
    fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data)
  })
    .then((response) => response.json())
    .then((json) =>{
      dispatch(fetchPosts())
    });
  }
}


export const deletePost = (id : number): AppThunk => {
  return (dispatch) => {
    fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
    method: 'DELETE',
    // body: JSON.stringify(id)
  })
    .then((response) => response.json())
    .then((json) =>{
      dispatch(fetchPosts())
    });
  }
}
