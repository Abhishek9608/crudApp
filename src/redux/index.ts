import { ThunkAction } from "redux-thunk";
import { configureStore } from "./configureStore";

// Redux store
export const store = configureStore();

export type RootState = ReturnType<typeof store.getState>;
export type Dispatch = typeof store.dispatch;
export type ActionType = ReturnType<typeof store.dispatch>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  ActionType
>;
