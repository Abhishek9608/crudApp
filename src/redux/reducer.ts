import { PostsAction } from "./types/action";
import { ActionTypes } from "./actionTypes";
import { post } from "../types";

interface stateType {
  posts: Array<post>
}

const initialState: stateType = {
  posts: []
};


function TodosReducer(
  state: Readonly<stateType> = initialState,
  action: PostsAction
): stateType {
  switch (action.type) {
    case ActionTypes.GET_POSTS:
      return {
        ...state,
        posts: action.posts
      };

    default:
      return state;
  }
}

export default TodosReducer;
