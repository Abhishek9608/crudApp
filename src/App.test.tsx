import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from './App'
import {store} from './redux'

it('App component snapshot should match correctly', () => {
    const tree = renderer
    .create(<Provider store={store}>
        <App/>
    </Provider>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  
});


