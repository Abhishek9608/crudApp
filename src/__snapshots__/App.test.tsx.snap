// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot should match correctly 1`] = `
<div>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    />
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        />
        <div
          className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          BLOGS
        </div>
        <div>
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="account of current user"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          />
        </div>
      </div>
    </header>
  </div>
  <div
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Add Post
    </button>
  </div>
  <div
    role="modal"
  />
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
    />
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        />
        <div
          className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          BLOGS
        </div>
        <div>
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="account of current user"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          />
        </div>
      </div>
    </header>
  </div>
  <div
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Add Post
    </button>
  </div>
  <div />
</div>
`;
